tests/gis_tests/geoapp/test_functions.py

```diff
@@ -233,6 +233,20 @@ def test_intersection(self):
                 expected = c.mpoly.intersection(geom)
             self.assertEqual(c.inter, expected)
 
+    @skipUnlessDBFeature("has_IsValid_function")
+    def test_isvalid(self):
+        valid_geom = fromstr('POLYGON((0 0, 0 1, 1 1, 1 0, 0 0))')
+        invalid_geom = fromstr('POLYGON((0 0, 0 1, 1 1, 1 0, 1 1, 1 0, 0 0))')
+
+        State.objects.create(name='valid', poly=valid_geom)
+        State.objects.create(name='invalid', poly=invalid_geom)
+
+        valid = State.objects.filter(name='valid').annotate(isvalid=functions.IsValid('poly')).first()
+        invalid = State.objects.filter(name='invalid').annotate(isvalid=functions.IsValid('poly')).first()
+
+        self.assertTrue(valid.isvalid)
```

Comment:
It would be safer to assertEqual with False/True since these assertions pass on any truthy/falsey value.